# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBuiltins.cpp
# Commit: 9a162cc369e3
# Full Hash: 9a162cc369e3b6f51f5767924e63d46539c4f631
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-02-22 09:14:45
# Regressor Bug: 1861078
# File Overlap Count: 1
# Description:
#   Bug 1861078 - wasm: make metadata offsets relative to CodeBlock->codeBase. r=rhunt
#   
#   Metadata used offsets to the beginning of code segment. Change that to use
#   offsets relative to the beginning of CodeBlock. This allows us to avoid offseting
#   metadata later on and prepare the work for simplification of stackmaps.
# ==============================================================================

diff -r e543a77786c9 -r 9a162cc369e3 js/src/wasm/WasmBuiltins.cpp
--- a/js/src/wasm/WasmBuiltins.cpp	Fri Feb 21 17:09:51 2025 +0000
+++ b/js/src/wasm/WasmBuiltins.cpp	Fri Feb 21 17:17:58 2025 +0000
@@ -686,7 +686,7 @@
     const wasm::Code& code, const uint8_t* pc, const CodeBlock** codeBlock) {
   const wasm::TryNote* tryNote = code.lookupTryNote((void*)pc, codeBlock);
   while (tryNote && tryNote->isDelegate()) {
-    pc = (*codeBlock)->segment->base() + tryNote->delegateOffset();
+    pc = (*codeBlock)->base() + tryNote->delegateOffset();
     const wasm::TryNote* delegateTryNote =
         code.lookupTryNote((void*)pc, codeBlock);
     MOZ_RELEASE_ASSERT(delegateTryNote == nullptr ||
@@ -847,8 +847,7 @@
 
         rfe->stackPointer =
             (uint8_t*)(rfe->framePointer - tryNote->landingPadFramePushed());
-        rfe->target =
-            codeBlock->segment->base() + tryNote->landingPadEntryPoint();
+        rfe->target = codeBlock->base() + tryNote->landingPadEntryPoint();
 
         // Make sure to clear trapping state if we got here due to a trap.
         if (activation->isWasmTrapping()) {